rust-sdk/whirlpool/src/config.rs

```diff
@@ -22,8 +57,19 @@ pub static WHIRLPOOLS_CONFIG_ADDRESS: Mutex<Pubkey> = Mutex::new(DEFAULT_WHIRLPO
 pub static WHIRLPOOLS_CONFIG_EXTENSION_ADDRESS: Mutex<Pubkey> =
     Mutex::new(DEFAULT_WHIRLPOOLS_CONFIG_EXTENSION_ADDRESS);
 
+/// Input type for setting the Whirlpools configuration.
+pub enum WhirlpoolConfigInput {
```

Comment:
nit:

Functionally the same, but instead of two enums you can do a single one:

```rust
pub enum WhirlpoolsConfigAddress {
  SolanaMainnet,
  SolanaDevnet,
  EclipseMainnet,
  EclipseTestnet
}

impl Into<Pubkey> for WhirlpoolsConfigAddress {
  pub fn into(val: self) -> Pubkey {
    ...
  }
```

Or can do it the other way around
```rust
pub enum WhirlpoolsConfigInput {
  Address(Pubkey)
  SolanaMainnet,
  SolanaDevnet,
  EclipseMainnet,
  EclipseTestnet
}

impl From<Pubkey> for WhirlpoolsConfigInput {
  fn from(val: Pubkey) -> Self {
    WhirlpoolsConfigInput::Address(val)
  }
}
```

This looks nicer for the caller as they can just do:
```rust
pub fn set_whirlpools_config_addres(address: Pubkey)

set_whirlpools_config_addres(WhirlpoolConfigInput::SolanaMainnet.into())
set_whirlpools_config_addres(Pubkey.from_str("..."))
```
Or if you choose option two:
```rust
pub fn set_whirlpools_config_addres(input: WhirlpoolConfigInput)

set_whirlpools_config_addres(WhirlpoolConfigInput::SolanaMainnet)
set_whirlpools_config_addres(Pubkey.from_str("...").into())
```

Instead, the current implementation results in:
```rust
set_whirlpools_config_addres(WhirlpoolConfigInput::Network(Network::SolanaMainnet))
set_whirlpools_config_addres(WhirlpoolConfigInput::Address(Pubkey.from_str("..."))

Reply:
This is nice! From a users perspective, I like the second option better, since I think most people will use Orca's Whirlpools when using the SDK.